NRLMMD-GEOIPS/geoips
# Add unit tests for good and bad plugins

# Reviewer Checklist
https://github.com/NRLMMD-GEOIPS/.github/blob/main/.github/review-template.md

# Related Issues
fixes NRLMMD-GEOIPS/geoips#[XXX]


# Testing Instructions

```
pip install -e .[test]
pytest ./tests
```

Should result in:
```
> pytest ./tests
====================================================================================================================================================== test session starts =======================================================================================================================================================
platform linux -- Python 3.10.10, pytest-7.2.2, pluggy-1.0.0
rootdir: /local/home/jsolbrig/geoips/github/geoips
plugins: cov-4.0.0, typeguard-2.13.3
collected 15 items                                                                                                                                                                                                                                                                                                               

tests/test_yaml_plugins.py ...............                                                                                                                                                                                                                                                                                 [100%]

======================================================================================================================================================== warnings summary ========================================================================================================================================================
geoips/interfaces/base.py:237
geoips/interfaces/base.py:237
geoips/interfaces/base.py:237
geoips/interfaces/base.py:237
  /local/home/jsolbrig/geoips/github/geoips/geoips/interfaces/base.py:237: DeprecationWarning: Use of 'entry_point_group' to specify the entry point group of an interface is deprecated. The interface name should match the name of the entry point group.
    warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================================================================================================= 15 passed, 4 warnings in 0.16s =================================================================================================================================================
```

# Summary


Testing Updates
===============

Add unit tests to validate yaml-based plugins and schema
--------------------------------------------------------
*From issue NRLMMD-GEOIPS/geoips#151: 2023-04-06, Add unit tests to validate plugins

::

    created: /geoips/tests/test_yaml_plugins.py
    modified: /geoips/pyproject.toml

# Output
